src/app.ts

```diff
@@ -176,7 +176,24 @@ app.on('ready', () => {
   webContents.on('new-window', (event: any, url, _1, _2, options) => {
     event.preventDefault()
 
-    if (/^(https:\/\/(mail|accounts)\.google\.com).*/.test(url)) {
+    if (/^https:\/\/accounts\.google\.com/.test(url)) {
+      mainWindow.loadURL(url)
+    } else if (/^https:\/\/mail\.google\.com/.test(url)) {
```

Comment:
Can you explain this logic to me?  It seems like when you switch accounts you shouldn't have any urls that point to the old account anymore?

Reply:
Do mean L181 or L179 + L181?

Reply:
L181+

Reply:
Don't know if I fully understand your question, but I'll try:

The first account you are signing in to is the default account. Every other will be an additional account. The default account is identified as `0` in `https://mail.google.com/mail/u/0/`. When you are switching accounts, the second account for example is `1` in ``https://mail.google.com/mail/u/<id>/`.

Reply:
I guess my question is more of when do links fall into this else statement? Which type of links are they? Notifications? mailto type links?

Reply:
Hm, I can't name all, but at least single email view, compose, pretty much everything(?). I only know of `accounts.google.com` being opened when you trying to add a new account.

Reply:
So, I played around with this and I have a more specific question.  When is this conditional true: `currentAccountId !== accountId`.  It seems like it never will be since all the links in the app will point to the current account.  So my question is are we adding this logic and it will never end up doing anything.

Reply:
Actually it does, maybe the code isn't clear enough. You need to be logged in with two accounts already.

Current behaviour:

![Kapture 2019-05-01 at 0 39 16](https://user-images.githubusercontent.com/8075074/57000054-d93fe880-6ba9-11e9-89d8-2304ffcaff41.gif)

Expected behaviour (this PR):

![Kapture 2019-05-01 at 0 40 52](https://user-images.githubusercontent.com/8075074/57000060-de9d3300-6ba9-11e9-8a2b-d83ebf56b91b.gif)

Reply:
That part I understand, but that is handled by this block

```ts
if (/^https:\/\/accounts\.google\.com/.test(url)) {
      mainWindow.loadURL(url)
}
```

I'm asking if this block is needed

```ts
} else if (/^https:\/\/mail\.google\.com/.test(url)) {
      const accountIdRegExp = /mail\/u\/(\d+)\//

      const currentAccountRes = accountIdRegExp.exec(
        mainWindow.webContents.getURL()
      )
      const currentAccountId = currentAccountRes && currentAccountRes[1]

      const accountRes = accountIdRegExp.exec(url)
      const accountId = accountRes && accountRes[1]

      if (currentAccountId !== accountId) {
        console.log(currentAccountId)
        console.log(accountId)

        mainWindow.loadURL(url)
        return
      }
```

Reply:
No, that's what I actually tried to explain 😄 

I've `console.log(url)` for you to show you what the URL for which action:

![Kapture 2019-05-01 at 0 50 44](https://user-images.githubusercontent.com/8075074/57000387-456f1c00-6bab-11e9-90e3-9fe080799566.gif)

See how the url is `https://mail.google.com/mail/u/1` for the second account. It will be `/u/0` for the default account when you switch back. `Add account` opens `https://accounts.google.com`.

Hope that clarifies it.

Reply:
Yes, that helps.  Sorry for the back and forth, but I thought it was the first if statement that would fire when switching accounts.

Reply:
No worries, I think I should add a comment to explain the logic.